File: Groove.SP/groove-sp-web/src/app/features/shipment/master-bill-of-lading/models/master-bill-of-lading-model.ts
import { BaseModel } from 'src/app/core/models/base.model';
import { ContractMasterModel } from './contract-master-model';

export interface MasterBillOfLadingModel extends BaseModel {
    id?: number;
    executionAgentId?: number;
    masterBillOfLadingNo: string;
    masterBillOfLadingType?: string;
    modeOfTransport?: string;
    movement?: string;
    carrierContractNo?: string;
    carrierName?: string;
    scac?: string;
    airlineCode?: string;
    vesselFlight?: string;
    vessel?: string;
    voyage?: string;
    flightNo?: string;
    placeOfReceipt?: string;
    portOfLoading?: string;
    portOfDischarge?: string;
    placeOfDelivery?: string;
    placeOfIssue?: string;
    issueDate: Date;
    onBoardDate: Date;

    /**
     * To define whether it is direct master or not
     *
     * + True, link directly to Shipment -> link to Shipment/Container via Consignments
     *
     * + False, link to House bill of lading -> link to Shipment/Container via BillOfLadingShipmentLoads
     */
    isDirectMaster: boolean;

    /**
     * For front-end only as editing Master bill of lading.
     * It links to Flight number for Air, else Vessel flight if Sea
     */
    vesselVoyage?: string;

    /**
     * To temporary set Itinerary Id linked to
     */
    itineraryId?: number;

    /**
     * To contain information on contract
     */
    contractMaster?: ContractMasterModel;

}

File: Groove.SP/Groove.SP.Core/Entities/MasterBillOfLadingModel.cs
using System;
using System.Collections.Generic;

namespace Groove.SP.Core.Entities
{
    public class MasterBillOfLadingModel : Entity
    {
        public long Id { get; set; }

        /// <summary>
        /// OrganizationID
        /// </summary>
        public long? ExecutionAgentId { get; set; }

        public string MasterBillOfLadingNo { get; set; }

        public string MasterBillOfLadingType { get; set; }

        public string ModeOfTransport { get; set; }

        public string Movement { get; set; }

        /// <summary>
        /// To map with <see cref="ContractMasterModel.CarrierContractNo"/> then show <see cref="ContractMasterModel.RealContractNo"/>
        /// </summary>
        public string CarrierContractNo { get; set; }

        public string CarrierName { get; set; }

        public string SCAC { get; set; }

        public string AirlineCode { get; set; }

        public string VesselFlight { get; set; }

        public string Vessel { get; set; }

        public string Voyage { get; set; }

        public string FlightNo { get; set; }

        public string PlaceOfReceipt { get; set; }

        public string PortOfLoading { get; set; }

        public string PortOfDischarge { get; set; }

        public string PlaceOfDelivery { get; set; }

        public string PlaceOfIssue { get; set; }

        public DateTime IssueDate { get; set; }

        public DateTime OnBoardDate { get; set; }

        /// <summary>
        /// To define whether it is direct master or not
        /// <list type="bullet">
        /// <item>True, link directly to Shipment -> link to Shipment/Container via Consignments</item>
        /// <item>False, link to House bill of lading -> link to Shipment/Container via BillOfLadingShipmentLoads</item>
        /// </list>
        /// </summary>
        public bool IsDirectMaster { get; set; }


        public virtual ICollection<MasterBillOfLadingContactModel> Contacts { get; set; }

        public virtual ICollection<MasterBillOfLadingItineraryModel> MasterBillOfLadingItineraries { get; set; }

        public virtual ICollection<ConsignmentItineraryModel> ConsignmentItineraries { get; set; }

        /// <summary>
        /// To define linkage between bill of lading and master bill of lading
        /// </summary>
        public virtual ICollection<BillOfLadingShipmentLoadModel> BillOfLadingShipmentLoads { get; set; }

        /// <summary>
        /// To define linkage between shipment and master bill of lading
        /// </summary>
        public virtual ICollection<ConsignmentModel> Consignments { get; set; }

        public virtual ICollection<BillOfLadingItineraryModel> BillOfLadingItineraries { get; set; }

        public ContractMasterModel ContractMaster { get; set; }
    }
}
